reports-backend/src/test/java/org/jboss/da/reports/backend/impl/VersionFinderTest.java

```diff
@@ -25,29 +30,148 @@
+
+    private static final String MULTI_BUILT_VERSION_RH_BEST = MULTI_BUILT_VERSION + ".redhat-18";
+
+    private static final String MULTI_BUILT_VERSION_RH4 = MULTI_BUILT_VERSION + ".redhat-16";
+
+    private static final String OTHER_RH_VERSION_1 = "1.0.0.redhat-1";
+
+    private static final String OTHER_RH_VERSION_2 = "1.0.0.redhat-18";
+
+    private static final String OTHER_RH_VERSION_3 = "1.1.1.redhat-15";
+
+    private static final String NON_OSGI_VERSION = "1.3";
+
+    private static final String NON_OSGI_VERSION_RHT = "1.3.0.redhat-4";
+
+    private static final String NON_OSGI_VERSION_2 = "1.3-Final";
+
+    private static final String NON_OSGI_VERSION_2_RHT = "1.3.0.Final-redhat-7";
+
+    private static final GA REQUESTED_GA = new GA("org.hibernate", "hibernate-core");
+
+    private static final GAV SOME_GAV = new GAV(REQUESTED_GA, "0.0.1");
+
+    private static final GAV NO_BUILT_GAV = new GAV(REQUESTED_GA, NO_BUILT_VERSION);
+
+    private static final GAV BUILT_GAV = new GAV(REQUESTED_GA, BUILT_VERSION);
+    private static final GAV BUILT_GAV_2 = new GAV(REQUESTED_GA, BUILT_VERSION_2);
+
+    private static final GAV MULTI_BUILT_GAV = new GAV(REQUESTED_GA, MULTI_BUILT_VERSION);
 
-    private List<String> builtHibernateVersions = Arrays.asList(BUILT_HIBERNATE_VERSIONS);
+    private static final GAV NON_OSGI_GAV = new GAV(REQUESTED_GA, NON_OSGI_VERSION);
+    private static final GAV NON_OSGI_GAV_2 = new GAV(REQUESTED_GA, NON_OSGI_VERSION_2);
+
+    private static final List<String> All_VERSIONS = Arrays.asList(OTHER_RH_VERSION_1,
+            OTHER_RH_VERSION_2, NO_BUILT_VERSION_2, NO_BUILT_VERSION, OTHER_RH_VERSION_3,
+            MULTI_BUILT_VERSION_RH2, BUILT_VERSION_RH, MULTI_BUILT_VERSION_RH1, BUILT_VERSION_2_RH,
+            MULTI_BUILT_VERSION_RH_BEST, MULTI_BUILT_VERSION_RH4, NON_OSGI_VERSION,
+            NON_OSGI_VERSION_RHT, BUILT_VERSION_2, NON_OSGI_VERSION_2, NON_OSGI_VERSION_2_RHT);
+
+    private static final List<String> BUILT_VERSIONS = Arrays.asList(OTHER_RH_VERSION_1,
+            OTHER_RH_VERSION_2, OTHER_RH_VERSION_3, MULTI_BUILT_VERSION_RH2, BUILT_VERSION_RH,
+            MULTI_BUILT_VERSION_RH1, BUILT_VERSION_2_RH, MULTI_BUILT_VERSION_RH_BEST, MULTI_BUILT_VERSION_RH4,
+            NON_OSGI_VERSION_RHT, NON_OSGI_VERSION_2_RHT);
+
+    private void prepare(List<String> versions) throws CommunicationException {
+        when(aproxConnector.getVersionsOfGA(REQUESTED_GA)).thenReturn(versions);
+    }
+
+    @Test
+    public void testVersionsForNonExistingGAV() throws CommunicationException {
+        prepare(null);
+        List<String> versions = versionFinder.getVersionsFor(SOME_GAV);
+        assertNull(versions);
+    }
+
+    @Test
+    public void testVersionsForGAV() throws CommunicationException {
+        prepare(All_VERSIONS);
+        List<String> versions = versionFinder.getVersionsFor(SOME_GAV);
+        assertNotNull(versions);
+        assertEquals(versions.size(), 8);
```

Comment:
Minor nitpick: could you revert the order?
```java
assertEquals(8, versions.size());
```

Otherwise I get that weird error message:
```
java.lang.AssertionError: expected:<9> but was:<8>
```

Should really be `expected:<8> but was <9>`